VideoLocker/src/main/java/org/edx/mobile/view/CourseDashboardFragment.java

```diff
@@ -71,17 +75,19 @@ public void onClick(View v) {
             }
         });
 
-        holder = createViewHolder(inflater, parent);
-
-        holder.typeView.setIcon(Iconify.IconValue.fa_comments_o);
-        holder.titleView.setText(R.string.discussion_title);
-        holder.subtitleView.setText(R.string.discussion_subtitle);
-        holder.rowView.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                environment.getRouter().showCourseDiscussionTopics(getActivity(), courseData);
-            }
-        });
+        if (courseData != null && !TextUtils.isEmpty(courseData.getCourse().getDiscussionUrl())) {
```

Comment:
>Seems like this should also be checking the DISCUSSIONS_ENABLED config flag?

yes it should.

>Was the issue about the config flag? Or are we also supposed to validate the discussion_url field (which does not seem to be actually being used) before enabling the discussion module?

`discussion_url` will be null for courses that use external discussion tools, otherwise not-null.

excerpt from the server ticket:
_Note: some courses use external discussion tools and the tab type for that would be ''external_discussion', which we do not support through our API._